assets/hip-1056/protobuf/stream/output/smart_contract_service.proto

```diff
@@ -129,33 +103,190 @@ message ContractSystemDeleteOutput {}
+ * 6. sender_id is not restated as it can be found and transformed from the RLP transaction
+ * 7. ContractID is not restated as it can be found in the RLP transaction or explicitly in the transactionbody input
+ */
+message EVMTransactionResult {
+     /**
+      * Result data from the function call.
+      * <p>
+      * This SHALL be encoded in RLP bytecode format.
+      */
+     bytes contract_call_result = 1;
 
-    /**
-    * Contract bytecode during deployment
-    */
-    bytes initcode = 3;
+     /**
+      * Any error message produced by the contract call. Max size of 100 bytes
+      * <p>
+      * This SHALL be unset if the contract call succeeded.
+      */
+     string error_message = 2;
 
-    /**
-    * Contract bytecode after deployment
-    */
-    bytes runtime_bytecode = 4;
+     /**
+      * A quantity of "gas" used.<br/>
+      * This represents the resource units expended to execute this
+      * contract call, and correlates to transaction costs.
+      */
+     uint64 gas_used = 3;
+}
+
+/**
+ * The slots read during the contract execution. These slots may or may not have been overwritten.
+ * The associated writes will be found in the state changes MapUpdateChange KVs
+ */
+message ContractSlotReads {
+     
+     message SlotRead {
+          oneof identifier {
+               /**
+               * The contract storage slot counter in this block 
+               * This is populated in place of the 256 bit word when the given slot is written to in state changes
+               */
+               int32 index = 1;
 
+               /**
+               * The key of this contratc storage slot, left padding of 0's by EVM is stripped to save space. 
+               * Indexers should left-pad with zeros to form a 256-bit word.
+               * This is populated when the slot was not written and only read
+               */
+               bytes key = 2;
+          }
+
+          /**
+           * The storage value in this slot, left padding of 0's by EVM is stripped to save space. 
+           * Indexers should left-pad with zeros to form a 256-bit word.
+          */
+          bytes read_value = 3;
+     }
+     
+     /**
+      * The contract assocaited with the storage slots this slot belongs to.
+      */
+     ContractID contractID = 1;
+
+     /**
+      * The storage slots that were read in this EVM exectuion. They may or may not have assocaited slot writes
+      */
+     repeated SlotRead slot_reads = 2;    
+}
+
+/**
+ * EVM transaction execution log storage details
+ * Details maps to the log object in eth_getTransactionReceipt response without repeating info already available in the input transaction
+ * Log bloom logic is removed as it may be calculated by block parser for both transactions and block level
+ */
+message EVMTransactionLog {
+     /**
+      * The contract emitting the log. ContractID vs 20 bye address is used to preserve space
+      */
+     ContractID contractID = 1;
```

Comment:
can remove contract_id, trace data and evm transaction (contract create / call / eth tx) has 1:1 mapping, so all logs in a EVMTraceData should come from the same contract id (can someone confirm?). we removed the contract id in the output (though the removal in EVMTransactionResult), we should also remove it here.

Reply:
I don't think that's the case. Contract `A` can call `B` and `C` and `B` and `C` can emit events so we'd need to disambiguate which contract it's coming from.
@lukelee-sl and @AlfredoG87?

Reply:
yeah, you are right. data in our db says so:

https://testnet.mirrornode.hedera.com/api/v1/contracts/results/logs?timestamp=1740795413.512890162

two logs for this ethereum transaction, for contract id 0.0.5508307 and 0.0.5505184